### PR TITLE
Improve release event types

### PR DESCRIPTION
**Pull request by mwiencek**
The only actual type change here is making `"release-events"` optional.

The rest of the changes are just clarifying that the `"date"` properties cannot be `null` (if they are, it's a bug), only `"area"`.

### CODE REVIEW

**Comment by mwiencek** [comment]
> The only actual type change here is making "release-events" optional.

I'd actually prefer we changed this in MBS at some point, though, by returning `"release-events": []` instead.

There's a similar issue for `"tracks"`.

File: api_types.ts [hunk 1]
```diff
@@ -22,7 +22,7 @@ export type IsoLanguageCode = string;
 /** ISO (four letter) code of a script. */
 export type IsoScriptCode = string;
 
-/** ISO 8601 `YYYY-MM-DD` date, can be partial (`YYYY-MM-DD` or `YYYY`). */
+/** ISO 8601 `YYYY-MM-DD` date, can be partial (`YYYY-MM-DD` or `YYYY`) or empty. */
```
Comment (kellnerd) [inline api_types.ts:25]:
```suggestion
/** ISO 8601 `YYYY-MM-DD` date, can be partial (`YYYY-MM` or `YYYY`) or empty. */
```
Hehe, I just realized that I was lazy here when I copy-pasted the date format string.
I have fixed this in 6b8bad08c7d1a8a9bbe3c24170e4723e57400b7 and removed the "or empty" from the type docstring again because in some places where this type alias is used it is `null` (date period) or optional instead of being empty.

The case where it is optional is "first-release-date" of recordings, which is inconsistent with "first-release-date" of release groups where it can only be empty.

**Review by kellnerd — APPROVED** [review]
Cool, thanks for checking these edge cases and finding test cases for them!

> The rest of the changes are just clarifying that the "date" properties cannot be null

No rule without an exception: `begin` and `end` date of date periods can be `null` 😁

> There's a similar issue for "tracks".

Good catch, I've addressed that in c44ce1496b607a03a6dc87432feea8806984c8b3.

File: api_types.ts (hunk 1 above)
Comment (mwiencek) [reply api_types.ts:25]: > No rule without an exception: begin and end date of date periods can be null 😁
>
> [...] in some places where this type alias is used it is null (date period) or optional instead of being empty.
>
> The case where it is optional is "first-release-date" of recordings, which is inconsistent with "first-release-date" of release groups where it can only be empty.

Thanks, I've tried to document those issues in [MBS-13519](https://tickets.metabrainz.org/browse/MBS-13519).
